### ISSUE TITLE
[Blazor] Allow only specific Components as Children

### ISSUE THREAD

**Issue by speyck** [dotnet/aspnetcore#39808]
I am currently working on a blazor class library. Let's say I have the following two components:

*Switch Component*
```cshtml
@typeparam T

@code {
    [Parameter]
    public T Evaluation { get; set; }

    [CascadingParameter]
    public List<Case<T>> Cases { get; set; }

    [Parameter]
    public RenderFragment Default { get; set; }
}
```
*Case Component*
```cshtml
@typeparam T

@code {
    [Parameter]
    public T Match { get; set; }


    [Parameter]
    public RenderFragment ChildContent { get; set; }

    [CascadingParameter]
    public List<Case<T>> Cases { get; set; }

    protected override void OnInitialized()
    {
        Cases.Add(this);
    }
}
```

What I want the markup usage of the Switch component to look like:
```cshtml
<Switch Evaluation="Input">
    <Case Match="@(4)">
        <a>four</a>
    </Case>
    <Case Match="@(0)">
        <a>zero</a>
    </Case>
    <Case Match="@(9)">
        <a>nine</a>
    </Case>
    <Default>
        <a>default</a>
    </Default>
</Switch>

@code {
    [Parameter]
    public int Input { get; set; }
}
```
The problem is that I get the following error:
> Unrecognized child content inside component 'Switch'. The component 'Switch' accepts child content through the following top-level items: 'Default'.

What I would like to achieve is, that the `Switch` component accepts both the `Default` parameter (how it's now) and any amount of `Case` components.

Also, a `Case` component should inherit the `typeparam` of the `Switch` component it's inside of.

---

In conclusion I would like to achieve the following things:
- `Case` component can only be used when inside a `Switch` component
- `Switch` component only takes `Default` renderfragment and `Case` components as children
- `Case` components inside `Switch` have same `typeparam`

Is any of the above possible?

---

.NET 6.0

**Comment by javiercn** [comment]
@speyck thanks for contacting us.

This is a dupe of https://github.com/dotnet/aspnetcore/issues/12302
